pages/[id].js

```diff
@@ -11,11 +11,20 @@ export default function Detail() {
   const { id } = router.query;
   const toast = useToast();
   const { data, isLoading } = useSWR(`api/roasts/${id}`, fetcher);
-
-  if (isLoading) return <h1>... is Loading</h1>;
-
+  const { data: favorites, isLoading: favoritesLoading } = useSWR(
+    () => (session ? "api/user" : null),
+    fetcher,
+    { refreshInterval: 1000 }
```

Comment:
Is there a reason why you poll for new data every second?
From the API endoint alone I'd say the user doesn't change. It would be different for comments on blog posts, which could be subject to change when there are other users. What's the rationale here?

Reply:
Without the refreshInterval it wont change the button to "Ist Favorit".

Reply:
Ah, I see. Have you tried using `mutate` after you toggle?

Reply:
Ah. Wenn man den Baum vor lauter Bäumen nicht mehr sieht. Klar!